components/ui/MultiselectEmail.tsx

```diff
@@ -103,16 +95,22 @@ function EmailInput({
         newValue: SingleValue<UserOption>,
         actionMeta: ActionMeta<UserOption>,
       ) => {
-        if (
-          actionMeta.action === "select-option" &&
-          newValue &&
-          isValidEmail(newValue.value)
-        )
-          submitEmail(newValue.value)
+        if (actionMeta.action === "select-option" && newValue?.value) {
+          if (isValidEmail(newValue.value)) {
+            submitEmail(newValue.value)
+          } else {
+            toast.error("Please enter a valid email")
+          }
+        }
       }}
       onKeyDown={(event: React.KeyboardEvent<HTMLElement>) => {
-        if (event.key === "Enter" && inputValue && isValidEmail(inputValue))
-          submitEmail(inputValue)
+        if (event.key === "Enter" && inputValue) {
+          if (isValidEmail(inputValue)) {
```

Comment:
Nit: this could be a function, which could reduce nesting and duplication if also used on line 99